x/merkledb/benchmarks/benchmark.go

```diff
@@ -0,0 +1,404 @@
+		return err
+	}
+	fmt.Printf("Completed initialization with hash of %v\n", rootID.Hex())
+	return nil
+}
+
+func resetRunningDatabaseDirectory() error {
+	runningDir := getRunningDatabaseDirectory()
+	if _, err := os.Stat(runningDir); err == nil {
+		err := os.RemoveAll(runningDir)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "unable to remove running directory : %v\n", err)
+			return err
+		}
+	}
+	err := os.Mkdir(runningDir, 0o777)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to create running directory : %v\n", err)
+		return err
+	}
+	err = CopyDirectory(getGoldenDatabaseDirectory(), runningDir)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to duplicate golden directory : %v\n", err)
+		return err
+	}
+	return nil
+}
+
+func runBenchmark() error {
+	levelDB, err := leveldb.New(
+		getRunningDatabaseDirectory(),
+		getLevelDBConfig(),
+		logging.NoLog{},
+		promRegistry,
+	)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to create levelDB database : %v\n", err)
+		return err
+	}
+
+	mdb, err := merkledb.New(context.Background(), levelDB, getMerkleDBConfig(promRegistry))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to create merkleDB database : %v\n", err)
+		return err
+	}
+	defer func() {
+		mdb.Close()
+		levelDB.Close()
+	}()
+
+	low := uint64(0)
+	var deleteDuration, addDuration, updateDuration, batchDuration time.Duration
+	batch := mdb.NewBatch()
+	for {
+		startBatchTime := time.Now()
+
+		// delete first 2.5k keys from the beginning
+		startDeleteTime := time.Now()
+		for keyToDeleteIdx := low; keyToDeleteIdx < low+databaseRunningBatchSize; keyToDeleteIdx++ {
+			entryHash := calculateIndexEncoding(keyToDeleteIdx)
+			err = batch.Delete(entryHash)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "unable to delete merkleDB entry : %v\n", err)
+				return err
+			}
+		}
+		deleteDuration = time.Since(startDeleteTime)
+		deleteRate.Set(float64(databaseRunningBatchSize) * float64(time.Second) / float64(deleteDuration))
+		deletesCounter.Add(databaseRunningBatchSize)
+
+		// add 2.5k past end.
+		startInsertTime := time.Now()
+		for keyToAddIdx := low + (*databaseEntries); keyToAddIdx < low+(*databaseEntries)+databaseRunningBatchSize; keyToAddIdx++ {
+			entryHash := calculateIndexEncoding(keyToAddIdx)
+			err = batch.Put(entryHash, entryHash)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "unable to insert merkleDB entry : %v\n", err)
+				return err
+			}
+		}
+		addDuration = time.Since(startInsertTime)
+		insertRate.Set(float64(databaseRunningBatchSize) * float64(time.Second) / float64(addDuration))
+		insertsCounter.Add(databaseRunningBatchSize)
+
+		// update middle 5k entries
+		startUpdateTime := time.Now()
+		for keyToUpdateIdx := low + ((*databaseEntries) / 2); keyToUpdateIdx < low+((*databaseEntries)/2)+databaseRunningUpdateSize; keyToUpdateIdx++ {
+			updateEntryKey := calculateIndexEncoding(keyToUpdateIdx)
+			updateEntryValue := calculateIndexEncoding(keyToUpdateIdx - ((*databaseEntries) / 2))
```

Comment:
This is incorrect, should be:
```suggestion
			updateEntryValue := calculateIndexEncoding(low)
```

Reply:
hmm.. I think that it's ok to use the `low` as you suggested, although using the above would yield different and unique values ( i.e. [low..low+5k] ).